hbase-backup/src/main/java/org/apache/hadoop/hbase/backup/impl/IncrementalTableBackupClient.java

```diff
@@ -262,6 +270,16 @@ private void updateFileLists(List<String> activeFiles, List<String> archiveFiles
   @Override
   public void execute() throws IOException {
     try {
+      Map<TableName, String> tablesToFullBackupIds = getFullBackupIds();
+
+      try (BackupAdminImpl backupAdmin = new BackupAdminImpl(conn)) {
```

Comment:
In the interest of allowing programmatic recovery form this case, I'd suggest to:
- check all tables before throwing an exception
- adapt `ColumnFamilyMismatchException` so it contains a field with the mismatched tables.

That way, users will be able to do:
```
try{
  createIncrementalBackup(...)
} catch (ColumnFamilyMismatchException e) {
  createFullBackup(e.mismatchedTables);
  createIncrementalBackup(...);
}
```

Reply:
Sure thing, do you think it makes sense to add the exception to the method signature?

Reply:
It does. I'm a big fan of fine-grained exception info at the method level (both in the javadoc and the `throws` statement).